test/vanilla/legacy/Expected/AcceptanceTests/BodyComplex/bodycomplex/aio/operations/_basic_operations.py

```diff
@@ -151,10 +164,15 @@ async def get_invalid(self, **kwargs: Any) -> "_models.Basic":
         error_map = {401: ClientAuthenticationError, 404: ResourceNotFoundError, 409: ResourceExistsError}
         error_map.update(kwargs.pop("error_map", {}))
 
+        _headers = case_insensitive_dict(kwargs.pop("headers", {}) or {})
```

Comment:
I know yesterday you didn't love that we had to convert to `case_insensitive_dict` in operations and in request builders. I think a good idea could be to only convert to `case_insensitive_dict` in the main operation if we need to, for example on line 121, we need case insensitive dict bc we're checking `api-version`. However, here we don't need a case insensitive dict bc we pass it directly to the request builder, which will convert it for us

If it makes autorest code too gross feel free to disregard, but I think it can be done neatly

Reply:
Yes @iscai-msft , this make lots sense on the generated code. I  modified two parts of logic to achieve this:
1) `model.Operation._imports_base` (for checking whether need to import `case_insensitive_dict`) 
2)  `serializer._OperationBaseSerializer.pop_kwargs_from_signature` (for checking whether need to convert pop outcomes to ``case_insensitive`)